tool/tsh/tsh.go

```diff
@@ -3233,6 +3236,12 @@ func makeClientForProxy(cf *CLIConf, proxy string, useProfileLogin bool) (*clien
 		cf.TracingProvider = tracing.NoopProvider()
 	}
 	c.Tracer = cf.TracingProvider.Tracer(teleport.ComponentTSH)
+	oldCtx := cf.Context
+	defer func() { cf.Context = oldCtx }()
+	cf.tracer = c.Tracer
+	ctx, span := c.Tracer.Start(cf.Context, "makeClientForProxy/init")
+	cf.Context = ctx
+	defer span.End()
```

Comment:
Is this all so `setClientWebProxyAddr` takes the ctx created here? Can we do this by adding `ctx` parameter where needed instead?

Reply:
True. I thought it was used for another function. Simplified